graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,331 @@
+        about voi, then VNviaSGM will return None.
+
+    order_seeds_subgraph: int, positive (default = 1)
+        Order used to  create induced subgraphs on `A`
+        and `B`. These subgraphs are centered about the seeds that were determined
+        by the subgraph generated by `order_voi_subgraph`. These two subgraphs will
+        be passed into the SGM algorithm.
+
+    n_init: int, positive (default = 100)
+        Number of restarts for soft seeded graph matching algorithm. Increasing the
+        number of restarts will make the probabilites returned more precise.
+
+    Attributes
+    ----------
+    n_seeds: int
+        Number of seeds passed in `seedsA` that occured in the induced subgraph about `voi`
+
+    nomination_list: 2d-array
+        An array containing vertex nominations in the form nomination list = [[j, p_val],...]
+        where p_val is the probability that the voi matches to node j in graph B (sorted by
+        descending probability)
+
+
+    References
+    ----------
+    .. [1] Patsolic, HG, Park, Y, Lyzinski, V, Priebe, CE. Vertex nomination via seeded graph matching. Stat Anal Data
+        Min: The ASA Data Sci Journal. 2020; 13: 229– 244. https://doi.org/10.1002/sam.11454
+
+
+
+    """
+
+    def __init__(self, order_voi_subgraph=1, order_seeds_subgraph=1, n_init=100):
+        if isinstance(order_voi_subgraph, int) and order_voi_subgraph > 0:
+            self.order_voi_subgraph = order_voi_subgraph
+        else:
+            msg = "order_voi_subgraph must be an integer > 0"
+            raise ValueError(msg)
+        if isinstance(order_seeds_subgraph, int) and order_seeds_subgraph > 0:
+            self.order_seeds_subgraph = order_seeds_subgraph
+        else:
+            msg = "order_seeds_subgraph must be an integer > 0"
+            raise ValueError(msg)
+        if isinstance(n_init, int) and n_init > 0:
+            self.n_init = n_init
+        else:
+            msg = "R must be an integer > 0"
+            raise ValueError(msg)
+
+    def fit(self, A, B, voi, seeds):
+        """
+        Fits the model to two graphs.
+
+        Parameters
+        ----------
+        A: 2d-array, square
+            Adjacency matrix of `A`, the graph where voi is known
+
+        B: 2d-array, square
+            Adjacency matrix of `B`, the graph where voi is not known
+
+        voi: int
+            Vertex of interest (voi)
+
+        seeds: list
+            List of length two `[seedsA, seedsB]` where first element is
+            the seeds associated with adjacency matrix A
+            and the second element the adjacency matrix associated with B, note
+            `len(seedsA)==len(seedsB)`. The elements of `seedsA` and `seedsB` are
+            vertices which are known to be matched, that is, `seedsA[i]` is matched
+            to vertex `seedsB[i]`.
+
+        Returns
+        -------
+        self: A reference to self
+        """
```

Comment:
you should add a check and test that voi is between 0 and len(A), and that it's an int

Reply:
take a look [here](https://github.com/microsoft/graspologic/blob/5bc9d10be6fffe77bc71fb23457515b3ebab50e7/graspologic/match/qap.py#L177). add any checks that you think are relevant to your function/might cause the code to break before it reaches gmp